internal/vault/cache_storage.go

```diff
@@ -49,7 +49,18 @@ const (
 	labelProviderNamespace    = "provider/namespace"
 )
 
-var EncryptionRequiredError = fmt.Errorf("encryption required")
+// commonMatchingLabels are included in all stored Clients. They are used for
+// nearly all client cache functions. Ideally they should never be altered,
+// since it could lead to orphaned cache entries and possibly other issues.
+var commonMatchingLabels = ctrlclient.MatchingLabels{
+	"app.kubernetes.io/name":       "vault-secrets-operator",
```

Comment:
Is this the right label, does the helm release name get appended to the app name?

Reply:
Yes, these should be the right labels for the client cache secrets. They are Helm agnostic, and should survive a potential re-installation of VSO etc.